rest/models/snapshot.go

```diff
@@ -69,6 +69,120 @@ func (p GetGainersLosersSnapshotParams) WithIncludeOTC(q bool) *GetGainersLosers
 	return &p
 }
 
+// GetOptionsChainParams is a set of parameters for the GetOptionsChainSnapshot method.
+type GetOptionsChainParams struct {
+	// The ticker for the option contract.
+	Ticker *string `validate:"required" path:"underlyingAsset"`
+
+	// The strike price of the option contract.
+	StrikePrice *string `query:"string_price"`
+
+	// The type of contract. Can be ContractTypeCall, ContractTypePut, or in some rare cases, ContractTypeOther.
+	ContractType *ContractType `query:"contract_type"`
+
+	// The contract's expiration date in YYYY-MM-DD format.
+	ExpirationDateEQ  *string `query:"expiration_date"`
+	ExpirationDateLT  *string `query:"expiration_date.lt"`
+	ExpirationDateLTE *string `query:"expiration_date.lte"`
+	ExpirationDateGT  *string `query:"expiration_date.gt"`
+	ExpirationDateGTE *string `query:"expiration_date.gte"`
+
+	Limit *int   `query:"Limit"`
+	Sort  *Sort  `query:"sort"`
+	Order *Order `query:"order"`
+}
+
+// NewGetOptionsChainParams will return a pointer to GetOptionsChainParams.
+func NewGetOptionsChainParams(ticker string) *GetOptionsChainParams {
+	return &GetOptionsChainParams{Ticker: &ticker}
+}
+
+// WithTicker sets underlying ticker symbol to params
+func (o *GetOptionsChainParams) WithTicker(ticker string) *GetOptionsChainParams {
```

Comment:
Other endpoints in this repo don't have the method receiver as a pointer.

Reply:
changed pointer receiver to non-pointer.